README.md

```diff
@@ -279,6 +279,31 @@ role.
 Similar to `controller` and `vlan`, the `parent` references the connection profile in
 the ansible role.
 
+#### `type: infiniband`
+
+For the infiniband connection, the following options are supported:
+
+- `p_key`: The infiniband P_Key to use for the device. The default value of `-1` means
+  to use the P_Key at the index 0. Otherwise, it is a 16-bit unsigned integer and the
+  high bit should be set if it is a "full membership" P_Key.
+- `transport_mode`: The ipoib (IP OVER INFINIBAND) connection operation mode. The
+  possible modes are `datagram` and `connected`.
+
+**Note:** For the `p_key` setting, initscripts provider will normalize the `p_key`
+  internally to always set up the full membership bit (e.g. 0x000a will be normalized
+  into 0x800a) while nm provider does not. And both providers calculate the interface
+  name exactly the same way - based on the `parent` interface name and the P_Key, as a
+  result, `interface_name` will vary for two providers. (e.g. if the parent
+  `interface_name` is `ib0` and user defined `p_key` is 0x000a, the `interface_name`
+  for nm provider should be `ib0.000a`and the `interface_name` for initscripts provider
+  should be `ib0.800a`)
```

Comment:
If the interface name is determined by the p_key, is there any reason for the user to specify both the p_key and the interface_name for infiniband interfaces if the mode is ipoib? Would it make sense for the role to calculate the interface_name instead?

Reply:
That is a very good point, we can omit the interface name as a setting and calculate the interface name internally.

Reply:
A new commit was added.